autogen/coding/local_commandline_code_executor.py

```diff
@@ -1,31 +1,45 @@
 from hashlib import md5
-import os
 from pathlib import Path
 import re
+from string import Template
 import sys
-import uuid
 import warnings
-from typing import ClassVar, List, Union
+from typing import Any, Callable, ClassVar, List, TypeVar, Union, cast
+from typing_extensions import ParamSpec
+from autogen.coding.func_with_reqs import FunctionWithRequirements, build_python_functions_file, to_stub
 
-from ..agentchat.agent import LLMAgent
-from ..code_utils import TIMEOUT_MSG, WIN32, _cmd, execute_code
+from ..code_utils import TIMEOUT_MSG, WIN32, _cmd
 from .base import CodeBlock, CodeExecutor, CodeExtractor, CommandLineCodeResult
 from .markdown_code_extractor import MarkdownCodeExtractor
 
 from .utils import _get_file_name_from_content
 
 import subprocess
 
+import logging
+
 __all__ = ("LocalCommandLineCodeExecutor",)
 
+A = ParamSpec("A")
+
 
 class LocalCommandLineCodeExecutor(CodeExecutor):
     SUPPORTED_LANGUAGES: ClassVar[List[str]] = ["bash", "shell", "sh", "pwsh", "powershell", "ps1", "python"]
+    FUNCTIONS_MODULE: ClassVar[str] = "functions"
+    FUNCTIONS_FILENAME: ClassVar[str] = "functions.py"
+    FUNCTION_PROMPT_TEMPLATE: ClassVar[
```

Comment:
@victordibia is this compatible with how autogen studio formulate the skill prompt?